### PR TITLE
[CORDA-446] Clean up other mentions of network map node and logic

### PR DESCRIPTION
**Pull request by al-r3**
Remove most remaining mentions of the network map node.

Update all the docs, to remove/update references to the network map

Also the NodeMessagingClient won't depend on the nodeReadyFuture anymore.
NodeMessagingClient won't have any special treatment for network map messages, as they don't exist anymore

### CODE REVIEW

File: docs/source/node-services.rst [hunk 1]
```diff
@@ -139,17 +139,10 @@ physical host and port information required for the physical
...
-nodes and advertised services in the Corda network. Only a single node 
-in the network (in future this will be a clustered service) should host 
-the NetworkMapService implementation. All other Corda nodes initiate 
-their remote connection to the ``NetworkMapService`` early in the 
-start-up sequence and wait to synchronise their local 
-``NetworkMapCache`` before activating any flows. For the 
+PersistentNetworkMapService
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ For the
```
Comment (shamsasari) [inline docs/source/node-services.rst]: incomplete sentence

File: node/src/main/kotlin/net/corda/node/services/messaging/NodeMessagingClient.kt [hunk 2]
```diff
@@ -267,19 +254,7 @@ class NodeMessagingClient(override val config: NodeConfiguration,
...
-     */
-    private fun makeP2PConsumer(session: ClientSession, networkMapOnly: Boolean): ClientConsumer {
-        return if (networkMapOnly) {
-            // Filter for just the network map messages.
-            val messageFilter = "hyphenated_props:$topicProperty like 'platform.network_map.%'"
-            session.createConsumer(P2P_QUEUE, messageFilter)
-        } else
-            session.createConsumer(P2P_QUEUE)
-    }
+    private fun makeP2PConsumer(session: ClientSession): ClientConsumer = session.createConsumer(P2P_QUEUE)
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/messaging/NodeMessagingClient.kt]: Don't really need a separate method for this now

File: node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt [hunk 3]
```diff
@@ -59,256 +33,23 @@ import javax.annotation.concurrent.ThreadSafe
...
-                                    override val sessionID: Long = random63BitValue()) : ServiceRequestMessage
-
-    @CordaSerializable
-    data class QueryIdentityResponse(val node: NodeInfo?)
-
-    // TODO Rename this RegistractionChangeRequest or similar (and related classes)
-    data class RegistrationRequest(val wireReg: WireNodeRegistration,
-                                   override val replyTo: SingleMessageRecipient,
-                                   override val sessionID: Long = random63BitValue()) : ServiceRequestMessage
+    val nodeRegistrations: MutableMap<PartyAndCertificate, NodeRegistrationInfo>
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: Why is this a MutableMap?

File: docs/source/node-services.rst (hunk 1 above)
Comment (al-r3) [reply docs/source/node-services.rst]: I've rephreased the paragraph.

File: node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt (hunk 3 above)
Comment (al-r3) [reply node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: Made it a map, everything seems to work.
For the record I think the whole NetworkMapService doesn't much do anything anyway.
It should either be deleted, or some logic currently in PersistentNetworkMapCache should be moved there

Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: Yes merge it with PersistentNetworkMapCache

Comment (al-r3) [reply node/src/main/kotlin/net/corda/node/services/network/NetworkMapService.kt]: hey, I'll look into doing it in a future PR, this one is already messy and bringing order to that class hierarchy will be even messier, I fear

**Pull request merged**